src/function/table/system/duckdb_log.cpp

```diff
@@ -0,0 +1,115 @@
+#include "duckdb/function/table/system_functions.hpp"
+
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/parser/parser.hpp"
+#include "duckdb/parser/parser_options.hpp"
+#include "duckdb/logging/log_manager.hpp"
+#include "duckdb/logging/log_storage.hpp"
+#include "duckdb/parser/tableref/subqueryref.hpp"
+
+namespace duckdb {
+
+struct DuckDBLogData : public GlobalTableFunctionState {
+	explicit DuckDBLogData(shared_ptr<LogStorage> log_storage_p) : log_storage(std::move(log_storage_p)) {
+		scan_state = log_storage->CreateScanEntriesState();
+		log_storage->InitializeScanEntries(*scan_state);
+	}
+	DuckDBLogData() : log_storage(nullptr) {
+	}
+
+	//! The log storage we are scanning
+	shared_ptr<LogStorage> log_storage;
+	unique_ptr<LogStorageScanState> scan_state;
+};
+
+static unique_ptr<FunctionData> DuckDBLogBind(ClientContext &context, TableFunctionBindInput &input,
+                                              vector<LogicalType> &return_types, vector<string> &names) {
+	bool context_id_only = false;
+	auto context_id_only_settings = input.named_parameters.find("context_id_only");
+	if (context_id_only_settings != input.named_parameters.end()) {
+		context_id_only = context_id_only_settings->second.GetValue<bool>();
+	}
+
+	if (context_id_only) {
+		names.emplace_back("context_id");
+		return_types.emplace_back(LogicalType::UBIGINT);
+	}
+
+	names.emplace_back("timestamp");
+	return_types.emplace_back(LogicalType::TIMESTAMP);
+
+	names.emplace_back("type");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	names.emplace_back("log_level");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	names.emplace_back("message");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	if (!context_id_only) {
+		names.emplace_back("scope");
+		return_types.emplace_back(LogicalType::VARCHAR);
+
+		names.emplace_back("client_context");
+		return_types.emplace_back(LogicalType::UBIGINT);
```

Comment:
There was an issue with `UBIGINT` being unwieldy when looking at it in the shell, please check